phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java

```diff
@@ -0,0 +1,314 @@
+        else {
+            try {
+                stmt.execute();
+                conn.commit();
+            } catch (SQLException e) {
+                throw new IOException(e);
+            }
+        }
+    }
+
+    private static  PreparedStatement setValuesToAddTaskPS(PreparedStatement stmt, PTable.TaskType taskType,
+            String tenantId, String schemaName, String tableName) throws SQLException {
+        stmt.setByte(1, taskType.getSerializedValue());
+        if (tenantId != null) {
+            stmt.setString(2, tenantId);
+        } else {
+            stmt.setNull(2, Types.VARCHAR);
+        }
+        if (schemaName != null) {
+            stmt.setString(3, schemaName);
+        } else {
+            stmt.setNull(3, Types.VARCHAR);
+        }
+        stmt.setString(4, tableName);
+        return stmt;
+    }
+
+    public static void addTask(PhoenixConnection conn, PTable.TaskType taskType, String tenantId, String schemaName,
+            String tableName, boolean accessCheckEnabled)
+            throws IOException {
+        PreparedStatement stmt = null;
+        try {
+            stmt = conn.prepareStatement("UPSERT INTO " +
+                    PhoenixDatabaseMetaData.SYSTEM_TASK_NAME + " ( " +
+                    PhoenixDatabaseMetaData.TASK_TYPE + ", " +
+                    PhoenixDatabaseMetaData.TENANT_ID + ", " +
+                    PhoenixDatabaseMetaData.TABLE_SCHEM + ", " +
+                    PhoenixDatabaseMetaData.TABLE_NAME + " ) VALUES(?,?,?,?)");
+            stmt = setValuesToAddTaskPS(stmt, taskType, tenantId, schemaName, tableName);
+        } catch (SQLException e) {
+            throw new IOException(e);
+        }
+        mutateSystemTaskTable(conn, stmt, accessCheckEnabled);
+    }
+
+    public static void addTask(PhoenixConnection conn, PTable.TaskType taskType, String tenantId, String schemaName,
+            String tableName, String taskStatus, String data, Integer priority, Timestamp endTs,
+            boolean accessCheckEnabled)
+            throws IOException {
+        PreparedStatement stmt = null;
+        try {
+            stmt = conn.prepareStatement("UPSERT INTO " +
+                    PhoenixDatabaseMetaData.SYSTEM_TASK_NAME + " ( " +
+                    PhoenixDatabaseMetaData.TASK_TYPE + ", " +
+                    PhoenixDatabaseMetaData.TENANT_ID + ", " +
+                    PhoenixDatabaseMetaData.TABLE_SCHEM + ", " +
+                    PhoenixDatabaseMetaData.TABLE_NAME + ", " +
+                    PhoenixDatabaseMetaData.TASK_STATUS + ", " +
+                    PhoenixDatabaseMetaData.TASK_PRIORITY + ", " +
+                    PhoenixDatabaseMetaData.TASK_END_TS + ", " +
+                    PhoenixDatabaseMetaData.TASK_DATA +
+                    " ) VALUES(?,?,?,?,?,?,?,?)");
+            stmt = setValuesToAddTaskPS(stmt, taskType, tenantId, schemaName, tableName);
+            if (taskStatus != null) {
+                stmt.setString(5, taskStatus);
```

Comment:
Any reason this logic shouldn't be in setValuesToAddTaskPS? (You could have an overload that took just the original 4 params that calls the full 8 param function)

Reply:
sure, will do